src/LuxCore.jl

```diff
@@ -179,6 +188,18 @@ function statelength(l::AbstractExplicitContainerLayer{layers}) where {layers}
     return sum(statelength, getfield.((l,), layers))
 end
 
+function stateless_apply(
+        model::AbstractExplicitContainerLayer{layers}, x, ps) where {layers}
+    if length(layers) == 1
+        layer_names = keys(getfield(model, layers[1]))
+    else
+        layer_names = layers
+    end
+    st = NamedTuple{layer_names}(NamedTuple() for _ in layer_names)
+
+    return first(apply(model, x, ps, st))
+end
```

Comment:
This won't work for nested settings. You will have to traverse the entire tree of layers to generate the states. For example, try this on `Chain(Dense(...), Parallel(+, Dense(...), Dense(...))`, and you will be the problem

Reply:
Ah, thanks!

I've added a `getstate` that recursively traverses the layers. Is it correct to assume that I encounter either a `NamedTuple` or an `AbstractExplicitLayer`? I saw the dispatch for `initialstates` with `l::Any` using `fmap` and I wasn't sure if I have to do that.